swarm/network/stream/stream.go

```diff
@@ -600,7 +601,11 @@ func (c *client) batchDone(p *Peer, req *OfferedHashesMsg, hashes []byte) error
 }
 
 func (c *client) close() {
-	close(c.next)
+	select {
```

Comment:
not safe against concurrent calls

Reply:
yes, but client.close() is called in peer.removeClient() which is under a lock, would protect this select? https://github.com/ethersphere/go-ethereum/blob/5347db84e2ca5d53b3ab9cd6077e02f67decaee8/swarm/network/stream/peer.go#L279

Reply:
I will trace why removeClient would be called multiple times, it should not happen for sure.